### PR TITLE
Disable failing tests on Mac

### PR DESCRIPTION
**Pull request by mspang**
#### Problem
The Mac build is broken.

 #### Summary of Changes
Disable tests that fail on Mac.

See #2721

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]
This seems ok as a stopgap, but we should presumably figure out why the tests fail eventually, right?

**Review by yunhanw-google — APPROVED** [review]
Thank you!

File: src/transport/tests/BUILD.gn [hunk 1]
```diff
@@ -49,7 +49,10 @@ chip_test_suite("tests") {
     "TestSecurePairingSession",
     "TestSecureSession",
     "TestSecureSessionMgr",
-    "TestTCP",
     "TestUDP",
   ]
+
+  if (current_os != "mac") {
+    tests += [ "TestTCP" ]
```
Comment (andy31415) [inline src/transport/tests/BUILD.gn:56]: Is there a GH issue to fix these tests? I would expect MAC to work with TCP

Comment (andy31415) [reply src/transport/tests/BUILD.gn:56]: Never mind - saw 2721 was mentioned.

**Review by andy31415 — APPROVED** [review]

**Comment by mspang** [comment]
> This seems ok as a stopgap, but we should presumably figure out why the tests fail eventually, right?

Yes. The bug is still open.

**Pull request merged**